components/form/events.md

```diff
@@ -0,0 +1,196 @@
+
+    EditContext myEditContext { get; set; }
+
+
+    private void OnSubmitHandler(EditContext editContext)
+    {
+        bool isFormValid = editContext.Validate();
+
+        if (isFormValid)
+        {
+            //apply some custom logic when the form is valud
+        }
+        else
+        {
+            //apply some custom logic when the form is not valid
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        myEditContext = new EditContext(person);
+        base.OnInitialized();
+    }
+
+    public class Person
+    {
+        [Editable(false)]
+        public int Id { get; set; }
+        [Required(ErrorMessage ="Add your first name")]
+        public string FirstName { get; set; }
+        [Required(ErrorMessage = "Add your last name")]
+        public string LastName { get; set; }
+        [Range(typeof(DateTime), "1/1/1900", "1/15/2020", ErrorMessage = "The Date of Birth must be between 1/1/1900 and 1/15/2021")]
+        public DateTime DOB { get; set; } = DateTime.Today.AddYears(-20);
+        public string CompanyName { get; set; }
+        public DateTime HireDate { get; set; }
+        public bool IsOnVacation { get; set; } = true;
+    }
+}
+````
+
+## OnValidSubmit
+
+The `OnValidSubmit` event fires when there are no validation erros in the Form. It is mapped to `OnValidEvent` event of the <a target="_blank" href="https://docs.microsoft.com/en-us/aspnet/core/blazor/forms-validation?view=aspnetcore-5.0#built-in-forms-components">Microsoft EditForm</a>
+
+>caption Use the OnValidSubmit event
+
+````CSHTML
+@* You can use the OnValidSubmit event to provide custom logic when the form is valid *@
+
+@using System.ComponentModel.DataAnnotations
+
+<TelerikForm EditContext="@myEditContext"
+             OnValidSubmit="@OnValidSubmitHandler">
+    <FormValidation>
+        <DataAnnotationsValidator></DataAnnotationsValidator>
+    </FormValidation>
+</TelerikForm>
+
+
+@code {
+    public Person person = new Person();
+
+    EditContext myEditContext { get; set; }
+
+
+    public void OnValidSubmitHandler()
+    {
+        //some logic when the form is valid.
+    }
+
+    protected override void OnInitialized()
+    {
+        myEditContext = new EditContext(person);
+        base.OnInitialized();
+    }
+
+    public class Person
+    {
+        [Editable(false)]
+        public int Id { get; set; }
+        [Required(ErrorMessage = "Add your first name")]
+        public string FirstName { get; set; }
+        [Required(ErrorMessage = "Add your last name")]
+        public string LastName { get; set; }
+        [Range(typeof(DateTime), "1/1/1900", "1/15/2020", ErrorMessage = "The Date of Birth must be between 1/1/1900 and 1/15/2021")]
+        public DateTime DOB { get; set; } = DateTime.Today.AddYears(-20);
+        public string CompanyName { get; set; }
+        public DateTime HireDate { get; set; }
+        public bool IsOnVacation { get; set; } = true;
+    }
+}
+````
+
+## OnInvalidSubmit
```

Comment:
According to MSDN (https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.editform.oninvalidsubmit?view=aspnetcore-5.0) this also can take the EditContext as a parameter. Test and document according to our behavior.

Again, check for the need to create an edit context in the first place.

Reply:
Done